presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -266,7 +266,7 @@ public BlockEncoding getEncoding()
     @Override
     public Block copyPositions(int[] positions, int offset, int length)
     {
-        checkValidPositions(positions, offset, length, positionCount);
+        checkValidPositionsArray(positions, offset, length);
```

Comment:
missing `checkValidPosition(positionCount, position);` below?

Reply:
`getId(position)` below does that